File: BnB-Mobile/lib/models/review.dart
class Review {
  final int id;
  final int userId;
  final int workerServiceId;
  final int rating;
  final String comment;
  final String userName;
  final DateTime createdAt;

  const Review({
    required this.id,
    required this.userId,
    required this.workerServiceId,
    required this.rating,
    required this.comment,
    required this.userName,
    required this.createdAt,
  });

  factory Review.fromJson(Map<String, dynamic> json) {
    return Review(
      id: json['id'],
      userId: json['user']?['id'] ?? 0,
      workerServiceId: json['worker_service_id'] ?? 0,
      rating: json['rating'],
      comment: json['comment'],
      userName: json['user']?['name'] ?? '',
      createdAt: DateTime.tryParse(json['created_at'] ?? '') ?? DateTime.now(),
    );
  }
}

File: BnB-Backend/app/Models/Review.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Attributes\Fillable;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

#[Fillable(['user_id', 'worker_service_id', 'rating', 'comment'])]
class Review extends Model
{
    use HasFactory, SoftDeletes;

    public function user()
    {
        return $this->belongsTo(User::class)->select(['id', 'name', 'phone']);
    }

    public function workerService()
    {
        return $this->belongsTo(WorkerService::class);
    }
}
